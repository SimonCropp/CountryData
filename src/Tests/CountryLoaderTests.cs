using CountryData;
using ObjectApproval;
using Xunit;

public class CountryLoaderTests
{
    [Fact]
    public void LoadLocationData()
    {
        var country = CountryLoader.LoadLocationData("PW");
        ObjectApprover.VerifyWithJson(country);
    }

    [Fact]
    public void LoadSpecificLocationData()
    {
        var country = CountryLoader.LoadPalauLocationData();
        ObjectApprover.VerifyWithJson(country);
    }

    [Fact]
    public void LoadAll()
    {
        CountryLoader.LoadAll();
        Assert.NotEmpty(CountryLoader.LoadedLocationData);
    }
}