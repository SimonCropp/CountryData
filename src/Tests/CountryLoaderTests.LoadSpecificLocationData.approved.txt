{
  "Name": "Palau",
  "Iso": "PW",
  "Iso3": "PLW",
  "IsoNumeric": 585,
  "Fips": "PS",
  "Capital": "Melekeok",
  "Area": 458.0,
  "Continent": "OC",
  "TopLevelDomain": ".pw",
  "CurrencyCode": "USD",
  "CurrencyName": "Dollar",
  "PhonePrefix": "680",
  "PostCodeFormat": "96940",
  "PostCodeRegex": "^(96940)$",
  "Languages": [
    "pau",
    "sov",
    "en-PW",
    "tox",
    "ja",
    "fil",
    "zh"
  ],
  "States": [
    {
      "Provinces": [
        {
          "Communities": [
            {
              "Places": [
                {
                  "Name": "Palau",
                  "PostCode": "96940",
                  "LatLong": "7.2257,134.3622"
                }
              ]
            }
          ]
        }
      ]
    }
  ]
}